### ISSUE TITLE
Entire chat history visible when joining new (private) channel

### ISSUE THREAD

**Issue by stefnats** [RocketChat/feature-requests#640]

**Comment by GoetheG** [comment]
Is there anthing new here? this is a serious compliance issue for me and my club. We cannot allow new users to see the chats there were in a channel before the user joined the channel.

**Comment by anuraag94** [comment]
Agreed, I'm a researcher using Slack as a platform to moderate small group interactions. To ensure the validity of my research, I need the history of all channels to be hidden from new users, including #general

**Comment by GoetheG** [comment]
@meghaParmar @anuraag94 @2betafactor @ManasN @johnyb0y @bussyjd @wargamesqcf and everybody else. Would you support a bounty on Bountysource for that with me?

**Comment by johnyb0y** [comment]
Hi @GoetheG To us it's really not that important. We've communicated this with our users and so far there have been no serious social incidents :)

**Comment by xmxax** [comment]
Hey there! I am also looking for this feature! 

Are there any updates or plans to actually implement this? 

Thank you for your hard work!

**Comment by Aens** [comment]
^ Let me know also if there's a way to do it.

**Comment by CaptWeiss** [comment]
I support this feature be added

**Comment by mrtnbm** [comment]
Yes, should be added

**Comment by suyon-won** [comment]
[For the REST users]
We can have a json['subscription']['ts'] from /api/v1/subscriptions.getOne.
And query /api/v1/groups.history with the 'oldest' option and ts.

(or custom logic for own app using /api/v1/groups.messages and ts)

**Comment by geekgonecrazy** [comment]
Can you try group dm?  If you add user here I believe creates new dm.  From what I can tell this still works exactly as we intend

**Comment by somoza** [comment]
November 2015, take your time ;)

**Comment by dirksan28** [comment]
After reading this thread - I guess there is still no such feature implemented right now... right?

**Comment by ktecv2000** [comment]
Any Update on this feature request?

**Comment by Navi2016** [comment]
Any update? Default should be not to allow new users in a channel to see history.

**Comment by Sidossi** [comment]
any news about this feature?
Thanks

**Comment by Branislav42** [comment]
Any update on this please?
